bin/worker.js

```diff
@@ -17,7 +18,9 @@ export default {
       return new Response('Missing required fields', { status: 400 })
     }
 
-    // TODO: Record retrieval stats to D1 asynchronously (do not block response)
-    return await retrieveFile(BASE_URL, pieceCid, env.CACHE_TTL)
+    const response = await retrieveFile(BASE_URL, pieceCid, env.CACHE_TTL)
+    const cacheStatus = response.headers.get('CF-Cache-Status')
```

Comment:
This header existing on `response` is an implementation detail of `retrieveFile`, and shouldn't be leaked to the outside. Could you change the function to return `{ response, cacheStatus }`?